hibernate-vector/src/main/java/org/hibernate/vector/AbstractOracleVectorJdbcType.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.vector;
+
+import java.sql.CallableStatement;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import org.hibernate.dialect.Dialect;
+import org.hibernate.dialect.OracleDialect;
+import org.hibernate.sql.ast.spi.SqlAppender;
+import org.hibernate.type.SqlTypes;
+import org.hibernate.type.descriptor.ValueBinder;
+import org.hibernate.type.descriptor.ValueExtractor;
+import org.hibernate.type.descriptor.WrapperOptions;
+import org.hibernate.type.descriptor.java.BasicPluralJavaType;
+import org.hibernate.type.descriptor.java.ByteJavaType;
+import org.hibernate.type.descriptor.java.JavaType;
+import org.hibernate.type.descriptor.java.PrimitiveByteArrayJavaType;
+import org.hibernate.type.descriptor.jdbc.ArrayJdbcType;
+import org.hibernate.type.descriptor.jdbc.BasicBinder;
+import org.hibernate.type.descriptor.jdbc.BasicExtractor;
+import org.hibernate.type.descriptor.jdbc.JdbcLiteralFormatter;
+import org.hibernate.type.descriptor.jdbc.JdbcType;
+import org.hibernate.type.descriptor.jdbc.internal.JdbcLiteralFormatterArray;
+
+import oracle.jdbc.OracleType;
+
+/**
+ * Specialized type mapping for generic vector {@link SqlTypes#VECTOR} SQL data type for Oracle.
+ * <p>
+ * This class handles generic vectors represented by an asterisk (*) in the format,
+ * allowing for different element types within the vector.
+ *
+ * @author Hassan AL Meftah
+ */
+public abstract class AbstractOracleVectorJdbcType extends ArrayJdbcType {
+
+	protected Class<?> javaType;
+
+	public AbstractOracleVectorJdbcType(JdbcType elementJdbcType, Class<?> javaType) {
+		super( elementJdbcType );
+		this.javaType = javaType;
+	}
+
+	public abstract String getFriendlyName();
+
+	public abstract void appendWriteExpression(String writeExpression, SqlAppender appender, Dialect dialect);
+
+	@Override
+	public int getJdbcTypeCode() {
```

Comment:
`getJdbcTypeCode()` is what we use for binding `null` and understanding the JDBC concept. What you want to change here is the SQL type code:

```suggestion
	public int getDefaultSqlTypeCode() {
```